fireant/slicer/widgets/highcharts.py

```diff
@@ -212,20 +210,20 @@ def _render_series(self, axis, axis_idx, axis_color, colors, data_frame_groups,
 
         :param axis:
         :param axis_idx:
-        :param axis_color:
         :param colors:
         :param data_frame_groups:
         :param render_series_label:
         :param references:
         :param is_timeseries:
         :return:
         """
-        has_multi_axis = 1 < len(self.items)
+        # Don't overwrite the iterator here, this function should keep advancing it.
+        _, axis_colors = itertools.tee(colors)
```

Comment:
This seems a bit odd. tee will make 2 new iterators that depend on `colors`. You then throw away 1. So `axis_colors = itertools.tee(colors, 1)` would have the same result.

Also, you're normally not supposed to advance the original iterator (`colors`) when you've used tee as all the values that have been yielded directly on that original iterator, will not end up in the new iterators. However, you don't seem to reuse `axis_colors` after its first use so I guess that's fine, just a bit confusing. Once you call next on `axis_colors` though, it will advance the original `colors` as well.

So understanding the desired behaviour here is a bit rough I think. If the desired behaviour is just peeking, it might be more readable to introduce the `more-itertools` package with its peek/spy options. If that's overkill you could also just get it from `colors`, and then chain it back on with itertools. That's apparently a top answer on StackOverflow.